Assets/Scripts/EnemyScripts/EnemyExplosion.cs

```diff
@@ -0,0 +1,50 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EnemyExplosion : MonoBehaviour
+{
+    public GameObject player;
+    private int explosionDamage;
+    public int ExplosionDamage
+    {
+        set { explosionDamage = value; }
+    }
+    private int explosionSize;
+    public int ExplosionSize
+    {
+        set
+        {
+            explosionSize = value;
+            transform.localScale = new Vector3(explosionSize, explosionSize, explosionSize);
+        }
+    }
+
+    private bool crit = false;
+    public bool Crit
+    {
+        set { crit = value; }
+    }
+    void Start()
+    {
+
+        ExplosionSize = 6;
+ 
+        StartCoroutine(DestroyExplosion());
+    }
+    private IEnumerator DestroyExplosion()
+    {
+        yield return new WaitForSeconds(0.25f);
+        Destroy(gameObject);
+    }
+
+    void OnTriggerEnter2D(Collider2D other)
+    {
+        if (other.gameObject.CompareTag("Player"))
+        {
+            other.gameObject.GetComponent<PlayerStats>().ReceiveDamage(10);
+            Destroy(gameObject);
```

Comment:
I would maybe not destroy the explosion here, as if you get hit by the explosion when it explodes, it gets destroyed instantly and you can't see the explosion that damaged you. The explosions are short enough that we probably won't need to worry about the player walking in and out of it and taking damage more than once, but if you wanted to fix that you could disable the collider when it damages the player (see melee enemy attack).